ARCHITECTURE.md

```diff
@@ -0,0 +1,54 @@
+
+## Chat types
+
+- The `Shop` chat is the primary 1:1 with a bot-user who wants to use webxdcs from the store.
+  Upon receiving a 1:1 chat message _or_ when a `DC::Contact` is verified with a QR-code, the bot creates
+  this kind of chat and sends the initial `store. xdc`. Currently implemented interaction is:
+    - Updating the store.xdc
+    - Receiving app updates
+    - Searching the store
+    - Downloading apps
+
+- The `Submit` chat is created by the bot when he receives a webxdc in the `shop`-chat. This chat handles the submission of one singular webxdc app, identified by its `app_id`. When this chat is created, the original submitter is added to the chat, the submitted webxdc is forwarded and a helper xdc (submit-helper.xdc) is sent to the chat.
+
+- The `Review` chat is created by clicking the submit button in the `submit-helper.xdc` (previous step). When creating such a chat, the bot chooses at most 3 testers and one publisher and adds them to the chat together with one forwarded instance of the submitted webxdc as well as the `review-helper.xdc`. When all tests are complete, the publisher can click on `publish` to finally release the webxdc to the store.
+
+- The `Genesis` chat type acts as an administrative group for the chat but at the moment it has no more functionality than letting users join the testers and publishers.
+
+In general, to add a webxdc as a developer without CLI access the three consecutive chat types are used: 
+1. `Shop`: To initiate a submission.
+2. `Submit`: To check all needed properties and start a review.
+3. `Review`: To test a submitted webxdc and finally release it to the store.
+
+To upgrade a webxdc it is intended to send a newer version to the `Submit` chat which will trigger a new
+release cycle.
+
+## App-Ids
+The bot uses two kinds of ids: One is the database's `row-id` and the other one is the `app-id` taken from 
+the webxdcs `manifest.toml`. Internally, only the `row-id` is used because `app-id` was just recently introduced and it's not possible to create a PRIMARY-KEY column of type TEXT in a SQLite database. The `row-id` is used as an unambiguous identifier for each `app-info` which is stored in the database.
+The newly added `app-id` should only be used to uniquely identify newly added webxdcs and their different versions. When adding a webxdc, the `app-id` is used to distinguish between app-upgrading requests and requests which initially add a new webxdc to the store. Other than that no more use cases are intended as of writing this.
+
+## Updating webxdcs
+It should be able to update webxdcs in two ways: 
```

Comment:
starting a section with "it" is generally to be avoided -- better choose a proper noun :)